src/types/mod.rs

```diff
@@ -2650,3 +2650,143 @@ mod transaction_fee_distribution {
+    // * In particular this means that the node is __not__ baking blocks.
+    ConsensusPassive,
+    // The node has been configured with baker keys however it is not currently baking and
+    // possilby never will.
+    NotInCommittee,
+    // The baker keys are registered however the baker is not in the committee
+    // for the current 'Epoch'.
+    AddedButNotActiveInCommittee,
+    // The node has been configured with baker keys that does not match the account.
+    AddedButWrongKeys,
+    // The node is member of the baking committee.
+    Baker(crate::types::BakerId),
+    // The node is member of the baking and finalization committee.
+    Finalizer(crate::types::BakerId),
+}
+
+#[derive(Debug)]
+/// The status of the requested node.
+pub struct NodeInfo {
+    // The version of the node.
+    // Note. if the returned version from the node does
+    // not conform to semantic versioning then the parsing will fail.
+    pub version:        semver::Version,
+    // Type of the node, see [NodeType]
+    pub node_type:      NodeType,
+    // The local (UTC) time of the node.
+    pub local_time:     chrono::DateTime<chrono::Utc>,
+    // How long the node has been alive.
+    pub uptime:         chrono::Duration,
+    // Information related to the network for the node.
+    pub network_info:   NetworkInfo,
+    // Information related to consensus for the node.
+    pub consensus_info: ConsensusInfo,
+}
+
+impl From<crate::v2::generated::node_info::NetworkInfo> for NetworkInfo {
+    fn from(network_info: crate::v2::generated::node_info::NetworkInfo) -> Self {
+        NetworkInfo {
+            node_id:             network_info.node_id,
+            peer_total_sent:     network_info.peer_total_sent,
+            peer_total_received: network_info.peer_total_received,
+            avg_bps_in:          network_info.avg_bps_in,
+            avg_bps_out:         network_info.avg_bps_out,
+        }
+    }
+}
+
+impl TryFrom<crate::v2::generated::NodeInfo> for NodeInfo {
+    type Error = anyhow::Error;
+
+    fn try_from(node_info: crate::v2::generated::NodeInfo) -> Result<Self, Self::Error> {
+        let version = semver::Version::parse(&node_info.peer_version)?;
+        let node_type = node_info.node_type.try_into()?;
+        let local_time = chrono::DateTime::<chrono::Utc>::from(std::time::UNIX_EPOCH)
+            + chrono::Duration::milliseconds(node_info.local_time.require()?.value as i64);
+        let uptime = DurationSeconds::from(node_info.peer_uptime.require()?.value).into();
+        let network_info = node_info.network_info.require()?.into();
+        let consensus_info = match node_info.consensus_info.require()? {
+            crate::v2::generated::node_info::ConsensusInfo::NoBaker(0) => ConsensusInfo::ConsensusNotRunning,
+            crate::v2::generated::node_info::ConsensusInfo::NoBaker(1) => ConsensusInfo::ConsensusPassive,
+            crate::v2::generated::node_info::ConsensusInfo::BakerConsensusInfo(baking_info) => {
```

Comment:
Please keep these tryfrom implementations in the v2 module.

Reply:
Moved to v2 mod.